scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -191,218 +152,60 @@ namespace <%= namespace %> {
-                            if (payment.ready()) {
-                                callback(null);
-                            } else {
-                                pollPayment(payment);
-                            }
-                        } else {
-                            HandleUserError(callback);
-                        }
-                    }
-                });
-            };
-
-            // Ensure we can checkout first
-            if (CurrentCheckout.ready()){
-                checkoutWithTokenizedPayment();
-            } else {
-                PollCheckoutReady((Checkout checkout, ShopifyError error) => {
-                    if (error != null) {
-                        callback(error);
-                    } else {
-                        checkoutWithTokenizedPayment();
-                    }
-                });
-            }
+            CheckoutMutator.CreateCheckout(
+                items: newLineItemInput,
+                callback: callback
+            );
         }
 
-        private void CheckoutUpdate(CompletionCallback callback) {
-            MutationQuery query = new MutationQuery();
-
+        private void UpdateCheckout(CompletionCallback callback) {
             // remove all line items them add them
-            List<string> lineItemsToRemove = CartLineItems.ConvertToLineItemIds(LineItems.All());
-            lineItemsToRemove.AddRange(DeletedLineItems);
-
-            List<CheckoutLineItemInput> lineItemsToAdd = CartLineItems.ConvertToCheckoutLineItemInput(LineItems.All());
-
-            DefaultQueries.checkout.LineItemsRemove(query, CurrentCheckout.id(), lineItemsToRemove);
-            DefaultQueries.checkout.LineItemsAdd(query, CurrentCheckout.id(), lineItemsToAdd);
-
-            Client.Mutation(query, (Mutation response, ShopifyError error) => {
-                if (error != null) {
-                    callback(error);
-                    return;
-                }
-
-                DeletedLineItems.Clear();
-
-                if (UpdateState(response.checkoutLineItemsAdd().checkout(), response.checkoutLineItemsAdd().userErrors())) {
-                    if (CurrentCheckout.ready()) {
-                        callback(null);
-                    } else {
-                        PollCheckoutAndUpdate(PollCheckoutReady, callback);
-                    }
-                } else {
-                    HandleUserError(callback);
+            List<string> toRemove = CartLineItems.ConvertToLineItemIds(LineItems.All());
+            toRemove.AddRange(DeletedLineItems);
+
+            List<CheckoutLineItemInput> toAdd = CartLineItems.ConvertToCheckoutLineItemInput(LineItems.All());
+
+            CheckoutMutator.UpdateItems(
+                checkout: CurrentCheckout,
+                lineItemsToAdd: toAdd,
+                lineItemsToRemove: toRemove,
+                callback: (e) => {
+                    DeletedLineItems.Clear();
+                    callback(e);
```

Comment:
If `UpdateItems` result in a failure, we still clear the deleted line items, do we want to do that?